test/blackbox-tests/test-cases/dune-cache/mode-hardlink.t

```diff
@@ -35,9 +36,9 @@ never built [target1] before.
   $ dune build --config-file=config target1 --debug-cache=shared,workspace-local \
   >   2>&1 | grep '_build/default/source\|_build/default/target'
   Workspace-local cache miss: _build/default/source: never seen this target before
-  Shared cache miss [3ad1761950da90e34e52b4c065db1504] (_build/default/source): not found in cache
+  Shared cache miss [e2f27218eba545659418b9c738e421ff] (_build/default/source): not found in cache
```

Comment:
Is the digest different because you bumped the version? If so, I'd recommend moving the version bump into a separate PR. Your feature on its own should not modify the digests, otherwise there's a bug somewhere.

Reply:
I assumed the hashes changed due to the change of the default value of `can_go_in_shared_cache` in https://github.com/ocaml/dune/pull/10710/commits/f9c0af97001967e08f634f3e76aa88df38b4d737 hence I promoted these changes.

But looking at it now I think the default should be `true` and just in the case of user rules it should be set to false (or whatever the configuration says) and they should, for the most part, flip back to the previous digests.

Reply:
Shouldn't the tests enable the caching of user rules because they rely on user rules? In which case, I would assume the digests to stay the same.

Reply:
You're right. I've bumped the version to 3.17 to be able use the `cache-user-rules` stanza, to be consistent with the tests using sexps to configure the cache but setting `DUNE_CACHE_USER_RULES=enabled` and making actions cacheable by default again restores the hashes to their previous values.